website/docs/en/plugins/rspack/html-rspack-plugin.mdx

```diff
@@ -318,3 +541,330 @@ module.exports = {
   ],
 };
 ```
+
+## Hooks
+
+HtmlRspackPlugin provides some hooks that allow you to modify tags or generated HTML code. The hooks object can be obtained through `HtmlRspackPlugin.getCompilationHooks`:
+
+```js title="rspack.config.js"
+const HtmlModifyPlugin = {
+  apply: function (compiler) {
+    compiler.hooks.compilation.tap('HtmlModifyPlugin', compilation => {
+      const hooks = HtmlRspackPlugin.getCompilationHooks(compilation);
+      // hooks.beforeAssetTagGeneration.tapAsync()
+      // hooks.alterAssetTags.tapAsync()
+      // hooks.alterAssetTagGroups.tapAsync()
+      // hooks.afterTemplateExecution.tapAsync()
+      // hooks.beforeEmit.tapAsync()
+      // hooks.afterEmit.tapAsync()
+    });
+  },
+};
+
+module.exports = {
+  //...
+  plugins: [new HtmlRspackPlugin(), HtmlModifyPlugin],
+};
+```
+
+### beforeAssetTagGeneration
+
+This hook will be called after collecting the assets from the compilation and generating the loading path, but before generating the tags.
+
+The `assets` can be modified here to add custom JS and CSS asset files.
+
+- **Type**: `AsyncSeriesWaterfallHook<[BeforeAssetTagGenerationData]>`
+- **Parameters**:
+  ```ts
+  type BeforeAssetTagGenerationData = {
+    assets: {
+      publicPath: string;
+      js: Array<string>;
+      css: Array<string>;
+      favicon?: string;
+    };
+    outputName: string;
+    plugin: {
+      options: HtmlRspackPluginOptions;
+    };
+  };
+  ```
+
+:::warning
+Only `assets.js`, `assets.css`, and `assets.favicon` can be modified. Modifications to other items will not take effect.
+:::
+
+The following code adds an additional `extra-script.js` and generates a `<script defer src="extra-script.js"></script>` tag in the final html content.
+
+```js title="rspack.config.js"
+const AddScriptPlugin = {
+  apply: function (compiler) {
+    compiler.hooks.compilation.tap('AddScriptPlugin', compilation => {
+      HtmlRspackPlugin.getCompilationHooks(
+        compilation,
+      ).beforeAssetTagGeneration.tapAsync(
```

Comment:
How about using `tapPromise` in the example, `tapPromise` can return the data instead of calling the `callback()`.